### PR TITLE
Update / add reprs for reference metadata objects

### LINKED ISSUE #2649: Add nice __repr__ s to the reference metadata objects
**Please describe the use case that requires this feature.**

The Reference metadata objects currently do not have reprs:

```
(Pdb) str(inputs[0]["reference_meta"].product)
'<ctapipe.io.metadata.Product object at 0x7f82685366d0>'
```

**Describe the solution you'd like**
Add reprs

### PR DESCRIPTION
**Pull request by maxnoe**
Fixes #2649

### CODE REVIEW

File: src/ctapipe/io/metadata.py [hunk 1]
```diff
@@ -172,6 +199,23 @@ def default_time(self):
...
 
+    def __repr__(self):
+        if self.stop_time is not None:
+            stop_time = f"'{self.stop_time.utc.isot}Z'"
+        else:
+            stop_time = None
+        return (
+            f"{self.__class__.__name__}("
+            f"name='{self.name}'"
+            f", id_='{self.id_}Z'"
```
Comment (Tobychev) [inline src/ctapipe/io/metadata.py:210]: why the "Z"?

File: src/ctapipe/io/metadata.py [hunk 2]
```diff
@@ -141,6 +159,15 @@ class Process(HasTraits):
     subtype = Unicode("")
     id_ = Unicode("")
 
+    def __repr__(self):
+        return (
+            f"{self.__class__.__name__}("
+            f"id_='{self.id_}Z'"
```
Comment (Tobychev) [inline src/ctapipe/io/metadata.py:165]: why the "Z"?

Comment (maxnoe) [reply src/ctapipe/io/metadata.py:165]: ISO8601 timezone code for UTC

File: src/ctapipe/io/metadata.py (hunk 1 above)
Comment (maxnoe) [reply src/ctapipe/io/metadata.py:210]: ISO8601 timezone code for UTC

File: src/ctapipe/io/metadata.py (hunk 2 above)
Comment (maxnoe) [reply src/ctapipe/io/metadata.py:165]: ah, wait, id. Good catch. thanks

Comment (maxnoe) [reply src/ctapipe/io/metadata.py:165]: copy / paste error

Comment (maxnoe) [reply src/ctapipe/io/metadata.py]: Fixed

**Review by Tobychev — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**